crates/module-system/module-implementations/sov-evm/tests/integration/transactions.rs

```diff
@@ -131,3 +131,69 @@ fn test_account_nonce() {
         }),
     });
 }
+
+// Check that if the same account deploys two contracts, each deployment results in a unique contract address
+#[test]
+fn test_deploy_many_contracts() {
+    let (mut runner, account, _) = setup();
+    let contract = SimpleStorageContract::default();
+    let contract_addr_1 = account.address().create(0);
+
+    // Deploy a contract and execute two transactions that update its storage.
+    {
+        let create_contract_tx_1 = create_deploy_tx(0, &contract, &account);
+        let set_value_tx_1 = create_set_arg_tx(5, 1, &contract, contract_addr_1, &account);
+        let set_value_tx_12 = create_set_arg_tx(7, 2, &contract, contract_addr_1, &account);
+
+        let evm = Evm::<S>::default();
```

Comment:
Define once outside

Reply:
EVM does not implement copy, so we either need to clone it or instantiate it again. I chose to instantiate it again since the module itself is stateless.